homeassistant/components/twinkly/light.py

```diff
@@ -0,0 +1,216 @@
+        """Return true if light is on."""
+        return self._is_on
+
+    @property
+    def brightness(self) -> Optional[int]:
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def state_attributes(self) -> dict:
+        """Return device specific state attributes."""
+
+        attributes = self._attributes
+
+        # Make sure to update any normalized property
+        attributes[ATTR_HOST] = self._client.host
+        attributes[ATTR_BRIGHTNESS] = self._brightness
+
+        return attributes
+
+    async def async_turn_on(self, **kwargs) -> None:
+        """Turn device on."""
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = int(int(kwargs[ATTR_BRIGHTNESS]) / 2.55)
+
+            # If brightness is 0, the twinkly will only "disable" the brightness,
+            # which means that it will be 100%.
+            if brightness == 0:
+                await self._client.set_is_on(False)
+                return
+
+            await self._client.set_brightness(brightness)
+
+        await self._client.set_is_on(True)
+
+    async def async_turn_off(self, **kwargs) -> None:
+        """Turn device off."""
+        await self._client.set_is_on(False)
+
+    async def async_update(self) -> None:
+        """Asynchronously updates the device properties."""
+        _LOGGER.info("Updating '%s'", self._client.host)
+
+        try:
+            self._is_on = await self._client.get_is_on()
+
+            self._brightness = (
+                int(round((await self._client.get_brightness()) * 2.55))
+                if self._is_on
+                else 0
+            )
+
+            device_info = await self._client.get_device_info()
+
+            if (
+                DEV_NAME in device_info
+                and DEV_MODEL in device_info
+                and (
+                    device_info[DEV_NAME] != self.__name
+                    or device_info[DEV_MODEL] != self.__model
+                )
+            ):
+                self.__name = device_info[DEV_NAME]
+                self.__model = device_info[DEV_MODEL]
+
+                if self._conf is not None:
+                    # If the name has changed, persist it in conf entry,
+                    # so we will be able to restore this new name if hass is started while the LED string is offline.
+                    self._hass.config_entries.async_update_entry(
+                        self._conf,
+                        data={
+                            CONF_ENTRY_HOST: self._client.host,  # this cannot change
+                            CONF_ENTRY_ID: self._id,  # this cannot change
+                            CONF_ENTRY_NAME: self.__name,
+                            CONF_ENTRY_MODEL: self.__model,
+                        },
+                    )
+
+            for key, value in device_info.items():
+                if key not in HIDDEN_DEV_VALUES:
+                    self._attributes[key] = value
```

Comment:
Please set the state attribute keys explicitly. We want to see what they are and the formatting.